lib/pengine/complex.c

```diff
@@ -596,16 +596,15 @@ pe__unpack_resource(xmlNode *xml_obj, pe_resource_t **rsc,
 
     if (id == NULL) {
         pe_err("Must specify id tag in <resource>");
-        return FALSE;
+        return pcmk_rc_schema_validation;
 
     } else if (rsc == NULL) {
         pe_err("Nowhere to unpack resource into");
-        return FALSE;
-
+        return EINVAL;
```

Comment:
A null destination pointer is an invalid argument.

Reply:
It would be worthwhile to do this check first, using CRM_CHECK() rather than pe_err(). pe_err() will log a configuration error whereas this would be a developer mistake.

Reply:
Done. `CRM_CHECK()` is used quite a bit for this situation. I would have expected `CRM_ASSERT()` if the intent is to catch a developer error.

Reply:
True, assert would be fine in this case